src/dsp_tools/utils/excel_to_json/properties.py

```diff
@@ -0,0 +1,457 @@
+        checks passed.
+
+    Raises:
+        UserError if any of the checks fail
+    """
+    mandatory_attributes = ["Spinbox", "List"]
+    mandatory_check = utl.col_must_or_not_empty_based_on_other_col(
+        check_df=check_df,
+        substring_list=mandatory_attributes,
+        substring_colname="gui_element",
+        check_empty_colname="gui_attributes",
+        must_have_value=True,
+    )
+    no_attributes = ["Checkbox", "Date", "Geonames", "Richtext", "TimeStamp"]
+    no_attribute_check = utl.col_must_or_not_empty_based_on_other_col(
+        check_df=check_df,
+        substring_list=no_attributes,
+        substring_colname="gui_element",
+        check_empty_colname="gui_attributes",
+        must_have_value=False,
+    )
+    # If neither has a problem, we return None
+    if mandatory_check is None and no_attribute_check is None:
+        return None
+    # If both have problems, we combine the series
+    elif mandatory_check is not None and no_attribute_check is not None:
+        final_series = pd.Series(np.logical_or(mandatory_check, no_attribute_check))
+    elif mandatory_check is not None:
+        final_series = mandatory_check
+    else:
+        final_series = no_attribute_check
+    # The boolean series is returned
+    return {"wrong gui_attributes": final_series}
+
+
+def _check_missing_values_in_row_raise_error(to_check_df: pd.DataFrame, excel_filename: str) -> None:
+    """
+    This function checks if all the required values are in the df. If all the checks are ok, the function ends without
+    any effect. If any of the checks fail, the called functions return a dictionary with the column name as key,
+    After all the checks are done, the a UserError is raised which contains the information in which column and row
+    there are problems.
+
+    Args:
+        to_check_df: pd.DataFrame that is to be checked
+        excel_filename: Name of the original Excel file
+
+    Raises:
+        UserError: if any of the checks are failed
+    """
+    # Every row in these columns must have a value
+    required_values = ["name", "super", "object", "gui_element"]
+    # If there are no problems, it returns an empty dict
+    missing_dict = utl.check_required_values(check_df=to_check_df, required_values_columns=required_values)
+    # This checks if the label columns have at least one value per row
+    missing_labels = utl.find_one_full_cell_in_cols(check_df=to_check_df, required_columns=language_label_col)
+    # If everything is ok, we get None, otherwise we update the dict
+    if missing_labels is not None:
+        missing_dict.update({"label": missing_labels})
+    # Some gui_element require a gui_attributes and others must not have one
+    missing_gui_attributes = _check_gui_attributes(check_df=to_check_df)
+    if missing_gui_attributes is not None:
+        missing_dict.update(missing_gui_attributes)
+    if missing_dict != {}:
```

Comment:
can be shortened to `if missing_dict`

(Sometimes I'm unsure about when I can apply this short way. To test, open a Python REPL and type `bool(xyz)`. This tests if you can use the shorthand `if xyz`, because `if xyz` is a shorthand for `if bool(xyz)`. 

P.S. While typing these lines, I was unsure if I were right. But Bing AI supports my view:

<img width="1099" alt="Screenshot 2023-08-29 at 11 49 21" src="https://github.com/dasch-swiss/dsp-tools/assets/39048939/8d75e7a4-8abb-4f7c-bdb9-dd192891ccb6">